.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build pull request
+
+on:
+  pull_request:
+    paths:
+      - ".github/workflows/pr-build.yml"
+      - "app/**"
+      - "gradle/**"
+      - "build.graddle.kts"
+      - "gradle.properties"
+      - "settings.gradle.kts"
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Make sure the release step uses its own credentials:
+          # https://github.com/cycjimmy/semantic-release-action#private-packages
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Build with Gradle
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew assembleRelease --no-daemon
+
+      - name: Sign APK
+        id: sign_apk
+        uses: ilharp/sign-android-release@v1
+        with:
+          releaseDir: build/app/outputs/apk/release
+          signingKey: ${{ secrets.SIGNING_KEYSTORE }}
+          keyStorePassword: ${{ secrets.SIGNING_KEYSTORE_PASSWORD }}
+          keyAlias: ${{ secrets.SIGNING_KEY_ALIAS }}
+          keyPassword: ${{ secrets.SIGNING_KEY_PASSWORD }}
+
+      - name: Upload build
+        uses: actions/upload-artifact@v3
+        with:
+          name: revanced-manager
+          path: ${{ steps.sign_apk.outputs.signedFile }}
```

Comment:
Need to be remove, PR can't access them.

Reply:
It needs to be signed with a key as release builds aren't signed by default. I suggested maybe debug keys.

Reply:
> It needs to be signed with a key as release builds aren't signed by default. I suggested maybe debug keys.

I guess we could sign the pr build with debug key.

Reply:
I do not know who has access to add keys, but I do not.

Reply:
> I guess we could sign the pr build with debug key, if that's the only option.

I guess technically the other option is that the unsigned build is uploaded and testers will need to sign it themselves before testing.

Reply:
@validcube We could also only compile debug builds, but debug builds for compose have performance issues, reducing the usefulness of testing it.